applications/finetune-quora-embeddings/benchmark.py

```diff
@@ -0,0 +1,354 @@
+    return [
+        InputExample(
+            texts=[
+                example["questions"]["text"][0],
+                example["questions"]["text"][1],
+            ],
+            label=int(example["is_duplicate"]),
+        )
+        for example in examples
+    ]
+
+
+@stub.function(
+    image=image, volumes={DATASET_DIR: DATASET_VOLUME}, gpu=GPU_CONFIG, timeout=1200
+)
+async def benchmark_mteb_model(model_name):
+    from datasets import load_from_disk
+    from sentence_transformers import util, SentenceTransformer
+    import numpy as np
+    import torch.nn as nn
+    import torch
+
+    dataset = load_from_disk(f"{DATASET_DIR}/{DATASET_NAME}")
+    test_dataset = dataset["test"]
+    model = SentenceTransformer(model_name)
+
+    sentence_to_embedding_map = dict()
+    batch_size = 5000
+    sentences = get_unique_sentences(
+        test_dataset, sentence_to_embedding_map, batch_size
+    )
+    embeddings = []
+    for item in sentences:
+        embeddings.extend(model.encode(item))
+
+    embeddings_tensor = torch.tensor(np.array(embeddings)).to("cuda")
+    # Create an embedding layer with pre-trained weights
+    embedding_layer = nn.Embedding.from_pretrained(embeddings_tensor)
+
+    sentence_pairs_elem_1, sentence_pairs_elem_2, labels = extract_unique_sentences(
+        test_dataset, sentence_to_embedding_map
+    )
+    sentence_pairs_embedding_1 = embedding_layer(
+        torch.as_tensor(sentence_pairs_elem_1, device="cuda")
+    )
+    sentence_pairs_embedding_2 = embedding_layer(
+        torch.as_tensor(sentence_pairs_elem_2, device="cuda")
+    )
+
+    cosine_scores = util.cos_sim(sentence_pairs_embedding_1, sentence_pairs_embedding_2)
+    predictions = np.diag(cosine_scores.cpu()).tolist()
+    return {name: f(labels, predictions) for name, f in EVALS.items()}
+
+
+@stub.function(
+    image=image,
+    volumes={DATASET_DIR: DATASET_VOLUME},
+    secret=Secret.from_name("openai"),
+    timeout=86400,
+    gpu=GPU_CONFIG,
+)
+async def benchmark_openai():
+    from datasets import load_from_disk
+    from sentence_transformers import util
+    from sklearn.metrics import roc_auc_score
+    import asyncio
+    import time
+    from tqdm import tqdm
+    from openai import AsyncOpenAI
+    import numpy as np
+    import torch
+    import torch.nn as nn
+
+    dataset = load_from_disk(f"{DATASET_DIR}/{DATASET_NAME}")
+    test_dataset = dataset["test"]
+    client = AsyncOpenAI()
+
+    sentence_to_embedding_map = dict()
+    batch_size = 128
+    num_sentences = 86235  # Number of unique sentences ( derived from len(sentences))
+
+    sem = asyncio.Semaphore(20)
+    sentences = get_unique_sentences(
+        test_dataset, sentence_to_embedding_map, batch_size
+    )
+    tqdm_monitoring_bar = tqdm(total=num_sentences)
+
+    async def embed_text(texts, progress_bar: tqdm):
+        async with sem:
+            response = await client.embeddings.create(
+                input=texts, model="text-embedding-ada-002"
```

Comment:
Move the model into the function parameter. 

def benchmark_openai(model="text-embedding-ada-002")

so you can map benchmark_openai.map(models) with the new models avialable